File: test_file.js
function calculateSum(a, b) {
    // This function adds two numbers and returns the result
    return a + b;
}

class Calculator {
    constructor(initialValue = 0) {
        this.value = initialValue;
    }
    
    add(x) {
        this.value += x;
        return this.value;
    }
    
    subtract(x) {
        this.value -= x;
        return this.value;
    }
}

/**
 * This is an existing docstring that might be outdated
 */
function complexFunction(data, filterValue = null, transform = true) {
    const result = [];
    for (const item of data) {
        if (filterValue === null || item > filterValue) {
            const processedItem = transform ? item * 2 : item;
            result.push(processedItem);
        }
    }
    return result;
}

File: fixtures/test_file.rs
fn calculate_sum(a: i32, b: i32) -> i32 {
    a + b
}

struct Calculator {
    value: i32,
}

impl Calculator {
    fn new(initial_value: i32) -> Self {
        Calculator { value: initial_value }
    }
    
    fn add(&mut self, x: i32) -> i32 {
        self.value += x;
        self.value
    }
    
    fn subtract(&mut self, x: i32) -> i32 {
        self.value -= x;
        self.value
    }
}

/// This is an existing docstring that might be outdated
fn complex_function(data: Vec<i32>, filter_value: Option<i32>, transform: bool) -> Vec<i32> {
    let mut result = Vec::new();
    for item in data {
        if filter_value.is_none() || item > filter_value.unwrap() {
            let processed_item = if transform { item * 2 } else { item };
            result.push(processed_item);
        }
    }
    result
}